### DISCUSSION TITLE
dyld: Library not loaded: /usr/local/opt/libusb/lib/libusb-1.0.0.dylib

### DISCUSSION THREAD

**Discussion by schmidsi** [foundry-rs/foundry#481] [Q&A] [answered]
After installing `foundryup` with

```bash
curl https://raw.githubusercontent.com/gakonst/foundry/master/foundryup/install | bash
source /Users/schmidsi/.zshrc
foundryup
```

Running `cast` or `forge` gives me this error:
```
dyld: Library not loaded: /usr/local/opt/libusb/lib/libusb-1.0.0.dylib
  Referenced from: /Users/schmidsi/.foundry/bin/forge
  Reason: image not found
[1]    13418 abort      forge
```

I'm using zsh on a Mac OS Big Sur 11.6.2 (20G314)

**Comment by schmidsi** [comment, marked as answer]
```
brew install libusb 
```

Did the trick

**Reply by bacoortuantu** [reply]
I love you

**Reply by naeemmaliki036** [reply]
Lav you 👌

**Comment by gakonst** [comment]
Nice catch - documented it here: https://github.com/gakonst/foundry/blob/master/README.md#error-when-running-forgecast

**Comment by dimitri-xyz** [comment]
Sorry if I am totally offbase here, but why is this library loaded in the first place? I didn't see any references in the code (current [head](https://github.com/foundry-rs/foundry/commit/3e262523b46149f2552323e0e749aa7843458444)) It seems it is a relic from previous work and it is only required by the cargo.lock file. Is there a branch that uses it? Is it an implicit library dependency?

**Reply by gakonst** [reply]
Yeah it comes from ethers to support hw wallet signing for cast. Haven't spent time trying to only load it if the hw wallets options are used unfortunately. Would be nice if that were the case.

**Reply by sambacha** [reply]
> Yeah it comes from ethers to support hw wallet signing for cast. Haven't spent time trying to only load it if the hw wallets options are used unfortunately. Would be nice if that were the case.

Can you make this a v1.0 priority ?
